src/fsharp/TypeChecker.fs

```diff
@@ -7635,6 +7635,11 @@ and TcComputationExpression cenv env overallTy mWhole interpExpr builderTy tpenv
         let mBuilderVal = mBuilderVal.MakeSynthetic()
         SynExpr.Lambda (false,false,SynSimplePats.SimplePats ([mkSynSimplePatVar false (mkSynId mBuilderVal builderValName)],mBuilderVal), runExpr, mBuilderVal)
 
+    let env =
+        match comp with     
+        | SynExpr.YieldOrReturn _ -> { env with eContextInfo = ContextInfo.ReturnInComputationExpression }
```

Comment:
@Tamil gave the following code sample.
![image](https://cloud.githubusercontent.com/assets/57396/15358196/5d3d8f82-1d03-11e6-9493-190aaa919aec.png)

So indeed we have a false positive. Adding as test case and trying to fix it.